nixos/modules/services/computing/slurm/slurm.nix

```diff
@@ -0,0 +1,405 @@
+          Acceptable values at present include "auth/none", "auth/authd", and "auth/
+          munge".
+        '';
+      };
+
+      cacheGroups = mkOption {
+        type = types.nullOr types.str;
+        default = "0";
+        example = null;
+        description = ''
+          If set to 1, the slurmd daemon will cache /etc/groups entries. This can
+          improve performance for highly parallel jobs if NIS servers are used and
+          unable to respond very quickly.
+        '';
+      };
+
+      cryptoType = mkOption {
+        type = types.nullOr types.str;
+        default = "crypto/munge";
+        example = null;
+        description = ''
+          The cryptographic signature tool to be used in the creation of job step
+          credentials. The slurmctld daemon must be restarted for a change in
+          CryptoType to take effect. Acceptable values at present include "crypto/
+          munge" and "crypto/openssl".
+        '';
+      };
+
+      returnToService = mkOption {
+        type = types.nullOr types.str;
+        default = "0";
+        example = null;
+        description = ''
+          Controls when a DOWN node will be returned to service.
+        '';
+      };
+
+      slurmctldPort = mkOption {
+        type = types.nullOr types.str;
+        default = "6817";
+        example = null;
+        description = ''
+          The port number that the SLURM controller, slurmctld, listens to for work.
+        '';
+      };
+
+      slurmdPort = mkOption {
+        type = types.nullOr types.str;
+        default = "6818";
+        example = null;
+        description = ''
+          The port number that the SLURM compute node daemon, slurmd, listens to for
+          work.
+        '';
+      };
+
+      switchType = mkOption {
+        type = types.nullOr types.str;
+        default = "switch/none";
+        example = null;
+        description = ''
+          Identifies the type of switch or interconnect used for application
+          communications.
+        '';
+      };
+
+      taskPlugin = mkOption {
+        type = types.nullOr types.str;
+        default = "task/none";
+        example = null;
+        description = ''
+          Identifies the type of task launch plugin, typically used to provide
+          resource management within a node (e.g. pinning tasks to specific
+          processors).
+        '';
+      };
+
+      inactiveLimit = mkOption {
+        type = types.nullOr types.str;
+        default = "0";
+        example = null;
+        description = ''
+          The interval, in seconds, after which a non-responsive job allocation
+          command (e.g. srun or salloc) will result in the job being terminated. If
+          the node on which the command is executed fails or the command abnormally
+          terminates, this will terminate its job allocation.
+        '';
+      };
+
+      killWait = mkOption {
```

Comment:
This is already by default 30 seconds if omitted from the configuration. The option can be be removed. And like this the others.